akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -274,8 +286,44 @@ object ByteString {
     def asByteBuffers: scala.collection.immutable.Iterable[ByteBuffer] = bytestrings map { _.asByteBuffer }
 
     def decodeString(charset: String): String = compact.decodeString(charset)
+
+    private[akka] def writeToOutputStream(os: OutputStream): Unit =
+      bytestrings.foreach(_.writeToOutputStream(os))
```

Comment:
Thinking a bit more about this, there could be a problem with the aggregated `ByteString` becoming very large. I mean 256 bytestrings of 1M each is fine but a 256M one might not be. Maybe there should be some kind of cap on how much to coalesce?